uber/models/__init__.py

```diff
@@ -0,0 +1,1306 @@
+                        try:
+                            value = datetime.strptime(value, c.DATE_FORMAT)
+                        except ValueError:
+                            value = dateparser.parse(value)
+                        value = value.date()
+                except Exception as error:
+                    log.debug(
+                        'Ignoring error coercing value for column {}.{}: {}',
+                        self.__tablename__, column.name, error)
+
+                setattr(self, column.name, value)
+
+        for column in self.__table__.columns:
+            if (not restricted or column.name in self.unrestricted) and (
+                    column.type is JSON or isinstance(column.type, JSON)):
+                fields = getattr(self, '_{}_fields'.format(column.name), {})
+                for field in fields.keys():
+                    if field in params:
+                        setattr(self, field, params[field])
+
+        if cherrypy.request.method.upper() == 'POST':
+            for column in self.__table__.columns:
+                if column.name in bools:
+                    setattr(
+                        self,
+                        column.name,
+                        bool(int(params.get(column.name, 0))))
+                elif column.name in checkgroups and column.name not in params:
+                    setattr(self, column.name, '')
+
+            if not ignore_csrf:
+                check_csrf(params.get('csrf_token'))
+
+        return self
+
+    def timespan(self, minute_increment=60):
+        def minutestr(dt):
+            return ':30' if dt.minute == 30 else ''
+
+        timespan = timedelta(minutes=minute_increment * self.duration)
+        endtime = self.start_time_local + timespan
+
+        startstr = self.start_time_local.strftime('%I').lstrip('0') + \
+            minutestr(self.start_time_local)
+
+        endstr = endtime.strftime('%I').lstrip('0') + \
+            minutestr(endtime) + endtime.strftime('%p').lower()
+
+        if self.start_time_local.day == endtime.day:
+            endstr += endtime.strftime(' %A')
+            if self.start_time_local.hour < 12 and endtime.hour >= 12:
+                return startstr + 'am - ' + endstr
+            else:
+                return startstr + '-' + endstr
+        else:
+            return startstr + self.start_time_local.strftime('pm %a - ') + \
+                endstr + endtime.strftime(' %a')
+
+
+# Make all of our model classes available from uber.models
+from uber.models.email import *  # noqa: F401,E402,F403
+from uber.models.core import *  # noqa: F401,E402,F403
+from uber.models.commerce import *  # noqa: F401,E402,F403
+from uber.models.promo_code import *  # noqa: F401,E402,F403
+from uber.models.admin import *  # noqa: F401,E402,F403
+from uber.models.tracking import *  # noqa: F401,E402,F403
+from uber.models.types import *  # noqa: F401,E402,F403
+
+# Exlplicitly import models used by the Session class to quiet flake8
```

Comment:
Typo in this comment.

Also, is doing stuff to pass flake8 going to be a Thing, or is this just because we're transitioning?

Reply:
In general this is not going to be a Thing<sup>TM</sup>. It's mostly because we're transitioning. Though there will always be exceptions when we have to do stuff to pass flake8. But not too much.